### PR TITLE
Authorization for BAP

### PR DESCRIPTION
**Pull request by shashank-on-codehunt**
BAP to Gateway 
and 
Gateway to BAP for on_search

### CODE REVIEW

File: local.env [hunk 1]
```diff
@@ -1,4 +1,6 @@
 BUYER_APP_PORT=2010
 BUYER_APP_ID=sample_mobility_bap
-BUYER_APP_URL=http://mobility-sample-bap:2010
-GATEWAY_URL=http://sample-gateway-registry:1010
\ No newline at end of file
+BUYER_APP_URL=http://localhost:2010
+GATEWAY_URL=http://localhost:1010
+privateKey=6os/UInKiuVpP5Vbu2m2PcVRX4SSIR2GbhrxQi1Wesum4lf4U7b24JgxjMDjQc1Srsc2/JgbuqkM7TKVS/SYdg==
```
Comment (hariharanweb) [inline local.env]: Lets use all caps for ENV variables

File: src/api/Api.js [hunk 2]
```diff
@@ -1,17 +1,24 @@
 import fetch from 'node-fetch';
 import LoggingService from '../services/LoggingService';
+import authHeader from '../utilities/SignVerify/AuthHeader';
 
-const doPost = (url, body) => {
+const doPost = async (url, bodyValue) => {
```
Comment (hariharanweb) [inline src/api/Api.js]: Lets rename to body - does not sound nice for person to read bodyValue

File: src/controllers/GenericResponse.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import Api from '../api/Api';
```
Comment (hariharanweb) [inline src/controllers/GenericResponse.js]: Should this file be in Controllers package, or move to some utils/filters/some logical package?

File: src/controllers/GenericResponse.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+const lookupUrl = `${process.env.GATEWAY_URL}/lookup`;
```
Comment (hariharanweb) [inline src/controllers/GenericResponse.js]: Make this as REGISTRY_URL (in our case its same), but we need capability to point our BAP to ONDC registry

File: src/controllers/GenericResponse.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      },
+    },
+  });
+};
+
+const sendErrorWithAuthorization = (res) => {
+  res.status(401).send('Error');
+};
+
+const getPublicKey = async (typeValue) => {
```
Comment (hariharanweb) [inline src/controllers/GenericResponse.js]: should we stick to `type` than `typeValue`

File: src/controllers/OnSearchController.js [hunk 6]
```diff
@@ -1,16 +1,21 @@
...
   logger.debug(`on_search called with ${JSON.stringify(req.body)}`);
-  SearchService.storeSearchResult(req.body);
-  res.send({
-    message: {
-      ack: {
-        status: 'ACK',
-      },
-    },
+
+  const publicKey = await genericResponse.getPublicKey(BecknGateway);
```
Comment (hariharanweb) [inline src/controllers/OnSearchController.js]: Can we move getPublicKey to something like LookupService.?

File: local.env (hunk 1 above)
Comment (priyajames29) [reply local.env]: Done in both .env and local.env

File: src/api/Api.js (hunk 2 above)
Comment (priyajames29) [reply src/api/Api.js]: Done.

File: src/controllers/GenericResponse.js (hunk 3 above)
Comment (priyajames29) [reply src/controllers/GenericResponse.js]: It has been moved to the utilities package

File: src/controllers/GenericResponse.js (hunk 4 above)
Comment (priyajames29) [reply src/controllers/GenericResponse.js]: Done.

File: src/controllers/GenericResponse.js (hunk 5 above)
Comment (priyajames29) [reply src/controllers/GenericResponse.js]: Done.

File: src/controllers/OnSearchController.js (hunk 6 above)
Comment (priyajames29) [reply src/controllers/OnSearchController.js]: Done.

File: src/services/LookUpService.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import Api from "../api/Api";
+
+const REGISTRY_URL = `${process.env.GATEWAY_URL}/lookup`;
```
Comment (hariharanweb) [inline src/services/LookUpService.js]: Make this process.env.REGISTRY_URL

**Pull request merged**